docs/ROADMAP.md

```diff
@@ -52,53 +52,58 @@ scenarios like limiting the customization and validation to a subset of
+very difficult to merge with upstream counterpart. This is forcing users to use setters
+and discouraging them from declaring other functions in the pipeline as they will be
+deleted during `kpt pkg update`. Merging pipeline correctly will reduce
+huge amount of friction in declaring new functions which in turn helps to avoid
+excessive parameterization. **Estimated completion date:** End of December 2021.
 
-### 10. Improve functionConfig UX
+### 12. Best practices for kpt with idiomatic package examples
 
-Package publishers want to provide command driven instructions (semi-automated)
-to configure a package that makes the user-guide/examples easy to follow.
-e.g. Porcelain to set values in functionConfig or generate functionConfig.
-**Estimated release date:** End of October 2021.
+We need to publish best practices to use kpt inorder to create and use kpt packages.
+This will help users to understand the right way of using kpt. These best practice
+guidelines should be backed by idiomatic kpt package examples. These packages should
+be designed reflecting the best practices, easily discoverable and simple to understand.
+This is an ongoing effort. For best practices guide **Estimated completion date:** End of November 2021.
+Idiomatic package examples **Estimated completion date:** End of December 2021.
 
-Package consumers should be able to detect configuration errors early before the
-package is assumed to be ready. e.g. Introduce validation for functionConfig.
-**Estimated release date:** End of November 2021.
+### 13. Explore various options for function runtime
 
-### 11. Improve Function Authoring Experience
+Currently, `kpt fn render` has dependency on docker to execute functions in pipeline.
+There are performance and docker dependency issues reported by customers. We will
+be exploring different function runtimes in order to address those issues. This is
+just an exploratory step and actual implementation if any, will be taken up after December 2021.
 
-We need a rich ecosystem of third party functions. Users should be able to write 
-functions with custom logic very quickly using the tools they are familiar with. 
-So we are investing on making function authoring experience very easy. This is an
-ongoing effort. [Starlark enhancements](https://github.com/GoogleContainerTools/kpt/issues/2504) 
-(and docs improvement) will be delivered by the end of September 2021. 
-**Estimated completion date:** End of December 2021.
+### 14. Integrate kpt with Cloud Code
+
+One of the major areas of investment is to integrate [Cloud Code](https://cloud.google.com/code) with kpt to provide
+package authoring assistance. Users can author Kptfile and functionConfig files with
+features like auto-complete and error detection. This significantly improves the
+discoverability of Kptfile schema, catalog functions and their functionConfigs.
+**Estimated release date:** Last week of September 2021 for Kptfile schema integration,
```

Comment:
It still has September 2021 ?

Reply:
let's get a quick PR to fix this.  Thanks for catching this Sunil!